p2p/peer.py

```diff
@@ -585,6 +585,15 @@ def __len__(self):
     def is_full(self) -> bool:
         return len(self) >= self.max_peers
 
+    def is_valid_connection_candidate(self, candidate: Node) -> bool:
+        # connect to no more then 2 nodes with the same IP
+        nodes_by_ip = groupby(
+            operator.attrgetter('remote.address.ip'),
```

Comment:
Just learned about the `operator` module. Pretty cool to have all operators available as functions :snake: